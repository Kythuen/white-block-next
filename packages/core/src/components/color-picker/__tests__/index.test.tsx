import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import ColorPicker from '../color-picker.vue'
// import { useColorPicker } from '../use'

describe('# ColorPicker', () => {
  describe('## Base', () => {
    it('### default config', () => {
      const wrapper = mount({
        render() {
          return <ColorPicker />
        }
      })
      // expect(wrapper.classes()).toEqual(['wb-color-picker'])
      // expect(wrapper.props()).toEqual({})
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  // describe('## Slots', () => {})

  // describe('## Events', () => {})

  // describe('## Props', () => {})

  // describe('## Composables', () => {})
})
