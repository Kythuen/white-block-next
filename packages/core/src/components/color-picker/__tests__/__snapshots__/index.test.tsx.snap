// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`# ColorPicker > ## Base > ### default config 1`] = `
<div
  bg="$wb-color-picker-background"
  class="wb-color-picker wb-color-picker--md"
  flex="~ col"
  gap="3"
  p="[calc(var(--wb-color-picker-padding)+0.125rem)]"
  ring="1 inset $wb-color-picker-border"
  rounded="$wb-color-picker-radius"
  select="none"
  w="$wb-color-picker-width"
>
  <div
    h="40"
    overflow="hidden"
    relative=""
    rounded="$comp-radius-sm"
    select="none"
    style="background: rgb(255, 0, 0);"
    w="full"
  >
    <div
      absolute=""
      h="full"
      inset="0"
      w="full"
    />
    <div
      absolute=""
      h="full"
      inset="0"
      w="full"
    />
    <div
      absolute=""
      right="-2"
      rounded="full"
      select="none"
      shadow="sm black"
      style="transform: translate3d(0px, 0px, 0);"
      top="-2"
    >
      <div
        cursor="pointer"
        h="4"
        ring="2 inset $wb-color-foreground"
        rounded="full"
        w="4"
      />
    </div>
  </div>
  <div
    flex=""
    gap="2"
    items="center"
  >
    <div
      flex="none"
      h="8"
      overflow="hidden"
      rounded="full"
      select="none"
      style="background: rgba(255, 255, 255, 0.4); background-size: 6px 6px;"
      w="8"
    >
      <div
        class="group"
        cursor="pointer"
        flex=""
        h="full"
        items="center"
        justify="center"
        overflow="hidden"
        style="background: rgba(0, 0, 0, 1);"
        text="white"
        w="full"
      >
        <i
          class="hidden group-hover:block"
          i="mdi-content-copy"
        />
      </div>
    </div>
    <div
      flex="~ col 1"
      gap="2.5"
      w="44"
    >
      <div
        h="2.5"
        relative=""
        select="none"
        w="full"
      >
        <div
          absolute=""
          flex=""
          h="full"
          inset="0"
          items="center"
          rounded="full"
          w="full"
        >
          <div
            bg="#f00"
            flex="none"
            h="full"
            rounded="l-2"
            w="2"
          />
          <div
            cursor="pointer"
            flex="1"
            h="full"
            select="none"
            w="full"
          />
          <div
            bg="#f00"
            flex="none"
            h="full"
            rounded="r-2"
            w="2"
          />
        </div>
        <div
          absolute=""
          cursor="pointer"
          h="4"
          left="0"
          rounded="full"
          select="none"
          shadow="sm black"
          style="background: rgb(0, 0, 0); transform: translate3d(0px, -50%, 0);"
          top="50%"
          w="4"
        >
          <div
            h="full"
            ring="2 inset $wb-color-foreground"
            rounded="full"
            w="full"
          />
        </div>
      </div>
      <div
        h="2.5"
        relative=""
        select="none"
        w="full"
      >
        <div
          absolute=""
          h="full"
          inset="0"
          overflow="hidden"
          rounded="full"
          style="background: rgba(255, 255, 255, 0.4); background-size: 6px 6px;"
          w="full"
        />
        <div
          absolute=""
          flex=""
          inset="0"
          items="center"
        >
          <div
            flex="none"
            h="full"
            rounded="l-2"
            style="background: rgb(0, 0, 0);"
            w="2"
          />
          <div
            cursor="pointer"
            flex="1"
            h="full"
            select="none"
            w="full"
          />
          <div
            flex="none"
            h="full"
            rounded="r-2"
            w="2"
          />
        </div>
        <div
          absolute=""
          cursor="pointer"
          h="4"
          left="0"
          rounded="full"
          select="none"
          shadow="sm black"
          style="background: rgb(0, 0, 0); transform: translate3d(0px, -50%, 0);"
          top="50%"
          w="4"
        >
          <div
            h="full"
            ring="2 inset $wb-color-foreground"
            rounded="full"
            w="full"
          />
        </div>
      </div>
    </div>
    <!--v-if-->
  </div>
  <!-- &lt;ColorPickerValue :selected-color="selectedColor" /&gt; -->
</div>
`;
